components/tidb_query_executors/src/runner.rs

```diff
@@ -286,8 +287,12 @@ pub fn build_executors<S: Storage + 'static>(
                 EXECUTOR_COUNT_METRICS.batch_limit.inc();
 
                 Box::new(
-                    BatchLimitExecutor::new(executor, ed.get_limit().get_limit() as usize)?
-                        .collect_summary(summary_slot_index),
+                    BatchLimitExecutor::new(
+                        executor,
+                        ed.get_limit().get_limit() as usize,
+                        is_scan_executor,
```

Comment:
Would you please explain this logic a little bit? Seems that `is_scan_executor` is always true as this point.

Reply:
`is_src_scan_executor` will be set as false when non-scan executor be created. (at the end of this for loop)